diff --git a/modules/programs/firefox.nix b/modules/programs/firefox.nix
index 1370d4c2..2142e112 100644
--- a/modules/programs/firefox.nix
+++ b/modules/programs/firefox.nix
@@ -10,21 +10,10 @@ let
 
   jsonFormat = pkgs.formats.json { };
 
-  mozillaConfigPath =
-    if isDarwin then "Library/Application Support/Mozilla" else ".mozilla";
-
-  firefoxConfigPath = if isDarwin then
-    "Library/Application Support/Firefox"
-  else
-    "${mozillaConfigPath}/firefox";
-
-  profilesPath =
-    if isDarwin then "${firefoxConfigPath}/Profiles" else firefoxConfigPath;
-
   nativeMessagingHostsPath = if isDarwin then
-    "${mozillaConfigPath}/NativeMessagingHosts"
+    "${cfg.mozillaConfigPath}/NativeMessagingHosts"
   else
-    "${mozillaConfigPath}/native-messaging-hosts";
+    "${cfg.mozillaConfigPath}/native-messaging-hosts";
 
   nativeMessagingHostsJoined = pkgs.symlinkJoin {
     name = "ff_native-messaging-hosts";
@@ -288,6 +277,45 @@ in {
         };
       };
 
+      mozillaConfigPath = mkOption {
+        type = types.str;
+        default = if isDarwin then
+          "Library/Application Support/Mozilla"
+        else
+          ".mozilla";
+        description = ''
+          Path used for add-ons.
+        '';
+      };
+
+      firefoxConfigPath = mkOption {
+        type = types.str;
+        default = if isDarwin then
+          "Library/Application Support/Firefox"
+        else
+          ".mozilla/firefox";
+        example = literalExpression ''
+          ".librewolf"
+        '';
+        description = ''
+          Path used for the config of firefox.
+        '';
+      };
+
+      profilesPath = mkOption {
+        type = types.str;
+        default = if isDarwin then
+          "Library/Application Support/Firefox/Profiles"
+        else
+          ".mozilla/firefox";
+        example = literalExpression ''
+          ".librewolf"
+        '';
+        description = ''
+          Path used for the profiles.
+        '';
+      };
+
       profiles = mkOption {
         type = types.attrsOf (types.submodule ({ config, name, ... }: {
           options = {
@@ -753,7 +781,7 @@ in {
     home.packages = lib.optional (cfg.finalPackage != null) cfg.finalPackage;
 
     home.file = mkMerge ([{
-      "${firefoxConfigPath}/profiles.ini" =
+      "${cfg.firefoxConfigPath}/profiles.ini" =
         mkIf (cfg.profiles != { }) { text = profilesIni; };
 
       "${nativeMessagingHostsPath}" = {
@@ -762,27 +790,27 @@ in {
         recursive = true;
       };
     }] ++ flip mapAttrsToList cfg.profiles (_: profile: {
-      "${profilesPath}/${profile.path}/.keep".text = "";
+      "${cfg.profilesPath}/${profile.path}/.keep".text = "";
 
-      "${profilesPath}/${profile.path}/chrome/userChrome.css" =
+      "${cfg.profilesPath}/${profile.path}/chrome/userChrome.css" =
         mkIf (profile.userChrome != "") { text = profile.userChrome; };
 
-      "${profilesPath}/${profile.path}/chrome/userContent.css" =
+      "${cfg.profilesPath}/${profile.path}/chrome/userContent.css" =
         mkIf (profile.userContent != "") { text = profile.userContent; };
 
-      "${profilesPath}/${profile.path}/user.js" = mkIf (profile.settings != { }
+      "${cfg.profilesPath}/${profile.path}/user.js" = mkIf (profile.settings != { }
         || profile.extraConfig != "" || profile.bookmarks != [ ]) {
           text =
             mkUserJs profile.settings profile.extraConfig profile.bookmarks;
         };
 
-      "${profilesPath}/${profile.path}/containers.json" =
+      "${cfg.profilesPath}/${profile.path}/containers.json" =
         mkIf (profile.containers != { }) {
           force = profile.containersForce;
           text = mkContainersJson profile.containers;
         };
 
-      "${profilesPath}/${profile.path}/search.json.mozlz4" = mkIf
+      "${cfg.profilesPath}/${profile.path}/search.json.mozlz4" = mkIf
         (profile.search.default != null || profile.search.privateDefault != null
           || profile.search.order != [ ] || profile.search.engines != { }) {
             force = profile.search.force;
@@ -929,7 +957,7 @@ in {
             '';
           };
 
-      "${profilesPath}/${profile.path}/extensions" =
+      "${cfg.profilesPath}/${profile.path}/extensions" =
         mkIf (profile.extensions != [ ]) {
           source = let
             extensionsEnvPkg = pkgs.buildEnv {
